New graphql schema (#662)
boostercloud/booster: docs/README.md

| includes | Object | Includes a given object |

Example:

```graphql
query {
  CartReadModels(filter: { itemsIds: { includes: "test-item" } }) {
    id
    price
    itemsIds
  }
}
```

_Note: Right now there is a limitation with complex properties in Arrays, where you just can filter them if you know the exact value of an element but is not possible to filter from a property of the element. As a workaround, you can use an array of ids of the complex property and filter for that property as in the example above._

##### Filter combinators

All the filters can be combined to create a more complex search on the same properties of the ReadModel.

| Filter |     Value     |                                      Description |
| :----- | :-----------: | -----------------------------------------------: |
| and    |   [Filters]   |       AND - all filters on the list have a match |
| or     |   [Filters]   | OR - At least one filter of the list has a match |
| not    | Filter/and/or |            The element does not match the filter |

Example:

```graphql
query {
  CartReadModels(filter: { or: [{ id: { contains: "a" } }, { id: { contains: "b" } }] }) {
    id
    price
    itemsIds
  }

> }

Reviewer: If it is supported, I would also add an example of combining `and` and `or` operators. Good documentation by the way!

Booster GraphQL API also provides support for real-time updates using subscriptions and websocket, to get more information about it go to the [GraphQL API](#subscribing-to-read-models) section.

#### Filtering a read model

> Booster GraphQL API provides support for filtering ReadModels on queries, subscriptions and at code level.

Reviewer 2: ```suggestion
The Booster GraphQL API provides support for filtering Read Models on queries, subscriptions and at code level.
```

> Using the GraphQL API endpoint you can retrieve the schema of your application so you can see what are the filters for every ReadModel and its properties. You can filter like this:

Reviewer 2: ```suggestion
Using the GraphQL API endpoint you can retrieve the schema of your application so you can see what are the filters for every Read Model and its properties. You can filter like this:
```

> Searching for a specific ReadModel by id

Reviewer 2: ```suggestion
Searching for a specific Read Model by `id`
```

```graphql
query {
  ProductReadModels(filter: { id: { eq: "test-id" } }) {
    id
    sku
    availability
    price
  }
}
```

#### Supported filters

> The currently supported filters are in the following ones. |

Reviewer 2: ```suggestion
The currently supported filters are the following ones:
```

##### Boolean filters

| Filter |   Value    |  Description |
| :----- | :--------: | -----------: |

> | eq     | true/false |     Equal to |
> | ne     | true/false | Not equal to |

Reviewer 2: ```suggestion
| eq     | boolean |     Equal to |
| ne     | boolean | Not equal to |
```

| Filter |  Value  |           Description |
| :----- | :-----: | --------------------: |
| eq     |  Float  |              Equal to |
| ne     |  Float  |          Not equal to |
| gt     |  Float  |          Greater than |
| ge     |  Float  | Greater or equal than |
| lt     |  Float  |            Lower than |
| le     |  Float  |   Lower or equal than |
| in     | [Float] | Exists in given array |

Example:

```graphql
query {
  ProductReadModels(filter: { price: { gt: 200 } }) {
    id
    sku
    availability
    price
  }
}
```

##### String filters

| Filter     |  Value   |                Description |
| :--------- | :------: | -------------------------: |
| eq         |  String  |                   Equal to |
| ne         |  String  |               Not equal to |
| gt         |  String  |               Greater than |
| ge         |  String  |      Greater or equal than |
| lt         |  String  |                 Lower than |
| le         |  String  |        Lower or equal than |
| in         | [String] |      Exists in given array |
| beginsWith |  String  | Starts with a given substr |
| contains   |  String  |    Contains a given substr |

Example:

```graphql
query {

>   ProductReadModels(filter: { sku: { begingsWith: "jelwery" } }) {

Reviewer 2: ```suggestion
  ProductReadModels(filter: { sku: { begingsWith: "jewelry" } }) {
```

    id
    sku
    availability
    price
  }
}
```

##### Array filters

| Filter   | Value  |             Description |
| :------- | :----: | ----------------------: |
| includes | Object | Includes a given object |

Example:

```graphql
query {
  CartReadModels(filter: { itemsIds: { includes: "test-item" } }) {
    id
    price
    itemsIds
  }
}
```

_Note: Right now there is a limitation with complex properties in Arrays, where you just can filter them if you know the exact value of an element but is not possible to filter from a property of the element. As a workaround, you can use an array of ids of the complex property and filter for that property as in the example above._

##### Filter combinators

All the filters can be combined to create a more complex search on the same properties of the ReadModel.
| Filter | Value | Description |
| :----- | :--------: | -----------: |
| and | [Filters] | AND - all filters on the list have a match |
| or | [Filters] | OR - At least one filter of the list has a match |

> | not | Filter/and/or | The element does not match the filter

Reviewer 2: This should be formatted probably
